src/components/KCheckbox/KCheckbox.vue

```diff
@@ -62,59 +62,36 @@
 </template>
 
 <script lang="ts" setup>
-import type { PropType } from 'vue'
-import { computed, useAttrs, useId, useSlots } from 'vue'
-import type { LabelAttributes } from '@/types'
+import { computed, useAttrs, useId } from 'vue'
+import type { CheckboxProps, CheckboxEmits, CheckboxSlots } from '@/types'
 import KLabel from '@/components/KLabel/KLabel.vue'
 import { CheckSmallIcon, IndeterminateSmallIcon } from '@kong/icons'
 import { KUI_ICON_SIZE_40 } from '@kong/design-tokens'
 
-const props = defineProps({
-  modelValue: {
-    type: Boolean,
-    default: false,
-    required: true,
-  },
-  label: {
-    type: String,
-    default: '',
-  },
-  labelAttributes: {
-    type: Object as PropType<LabelAttributes>,
-    default: () => ({}),
-    validator: (value: LabelAttributes): boolean => {
```

Comment:
Let's keep this in a watcher like https://github.com/Kong/kongponents/blob/40910f9c647d22a621475b1c72a5a3cc26650d6f/src/components/KButton/KButton.vue#L33-L38